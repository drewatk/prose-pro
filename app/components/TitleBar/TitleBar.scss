.title-bar {
  background-color: #4f4f4f;
  color: white;
  padding: 0.3rem 0.3rem;
  -webkit-user-select: none;

  &-darwin {
    -webkit-app-region: drag;

    & > &-buttons-placeholder {
      display: inline-block;
      width: 75px;
      height: inherit;
    }
  }

  &-header {
    display: inline-block;
  }

  &-header {
    margin-top: 0.4rem;
    margin-bottom: 0;
    font-size: 1.25rem;

    :global(.text-muted) {
      color: #ddd !important;
      margin-left: 0.5rem;
    }
  }

  :global(.nav-item) {
    font-size: 0.9rem;

    a {
      color: lighten(#007bff, 25%);
    }

    a:hover {
      color: lighten(#0056b3, 25%);
    }
  }
}
