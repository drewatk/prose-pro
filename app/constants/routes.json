{
  "HOME": "/",
  "PROJECT_SETUP": "/ProjectSetup"
}
