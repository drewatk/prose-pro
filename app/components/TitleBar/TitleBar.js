/**
 * DISABLED TEST FILE
 */

import React from "react";
import { connect } from "react-redux";
import { Nav, NavItem, NavLink } from "reactstrap";
import { platform } from "os";
import routes from "app/constants/routes.json";
import { toggleShowHistory, toggleShowFileList } from "app/actions/view";
import { push } from "connected-react-router";
import styles from "./TitleBar.scss";
import { deSelectFile } from "app/actions/file_selection";
import { deSelectProject } from "app/actions/project_selection";

// True if on OSX
const darwin = platform() === "darwin";

export const TitleBar = props => {
  const { title, pathname, onFilesClick, onHistoryClick, onBackClick } = props;

  return (
    <div
      className={`${styles.titleBar} ${
        darwin ? styles.titleBarDarwin : ""
      } d-flex`}
    >
      {/* Insert a placeholder to offset for the buttons if we're on OSX */}
      <span className={styles.titleBarDarwinButtonsPlaceholder} />
      <h1 className={styles.titleBarHeader}>
        ProsePro
        <small className="text-muted">{title}</small>
      </h1>

      {pathname === routes.EDITOR && (
        <Nav className="ml-auto">
          <NavItem>
            <NavLink
              id="back-button"
              onClick={() => {
                onBackClick();
              }}
              href="#"
            >
              <i className="fas fa-arrow-left" />
              &nbsp;Back
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              id="files-button"
              onClick={() => {
                onFilesClick();
              }}
              href="#"
            >
              <i className="far fa-file" />
              &nbsp;Files
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink
              id="history-button"
              onClick={() => {
                onHistoryClick();
              }}
              href="#"
            >
              <i className="fas fa-history" />
              &nbsp;History
            </NavLink>
          </NavItem>
        </Nav>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  pathname: state.router.location.pathname
});

const mapDispatchToProps = dispatch => {
  return {
    onFilesClick: () => dispatch(toggleShowFileList()),
    onHistoryClick: () => dispatch(toggleShowHistory()),
    onBackClick: () => {
      dispatch([deSelectFile(), deSelectProject(), push(routes.PROJECT_SETUP)]);
    }
  };
};

const WithTitleBar = connect(
  mapStateToProps,
  mapDispatchToProps
)(TitleBar);

export default WithTitleBar;
