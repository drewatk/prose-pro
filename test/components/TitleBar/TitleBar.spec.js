import React from "react";
import Enzyme, { shallow } from "enzyme";
// import { platform } from "os";
import Adapter from "enzyme-adapter-react-16";
import routes from "app/constants/routes.json";
import { TitleBar } from "app/components/TitleBar/TitleBar";

Enzyme.configure({ adapter: new Adapter() });

// Todo: this seems innefective
jest.mock("os");

describe("TitleBar", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<TitleBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (on editor page)", () => {
    const pathname = routes.EDITOR;

    const wrapper = shallow(<TitleBar pathname={pathname} />);
    expect(wrapper).toMatchSnapshot();
  });

  // TODO: Unable to mock os module correctly
  // it("Doesn't show span on Windows", () => {
  //   platform.mockReturnValueOnce("win32");
  //   const wrapper = shallow(<TitleBar />);
  //   expect(wrapper.find(".titleBarDarwin").exists()).toBeFalsy();
  // });

  // it("Doesn't shows span on OSX", () => {
  //   platform.mockReturnValueOnce("darwin");

  //   console.log(platform);
  //   const wrapper = shallow(<TitleBar />);
  //   expect(wrapper.find(".titleBarDarwin").exists()).toBeTruthy();
  // });

  it("Displays nav on Editor route", () => {
    const pathname = routes.EDITOR;

    const wrapper = shallow(<TitleBar pathname={pathname} />);

    expect(wrapper.find("Nav").exists()).toBeTruthy();
  });

  it("calls onFilesClick", () => {
    const onFilesClick = jest.fn();
    const pathname = routes.EDITOR;

    const wrapper = shallow(
      <TitleBar pathname={pathname} onFilesClick={onFilesClick} />
    );

    wrapper.find("#files-button").simulate("click");
    expect(onFilesClick).toHaveBeenCalledTimes(1);
  });

  it("calls onHistoryClick", () => {
    const onHistoryClick = jest.fn();
    const pathname = routes.EDITOR;

    const wrapper = shallow(
      <TitleBar pathname={pathname} onHistoryClick={onHistoryClick} />
    );

    wrapper.find("#history-button").simulate("click");
    expect(onHistoryClick).toHaveBeenCalledTimes(1);
  });
});
