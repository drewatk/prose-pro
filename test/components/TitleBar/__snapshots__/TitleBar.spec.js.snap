// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar matches snapshot 1`] = `
<div
  className="titleBar  d-flex"
>
  <span
    className="titleBarDarwinButtonsPlaceholder"
  />
  <h1
    className="titleBarHeader"
  >
    ProsePro
    <small
      className="text-muted"
    />
  </h1>
</div>
`;
